internal/langserver/handlers/command/module_calls.go

```diff
@@ -48,72 +49,41 @@ func (h *CmdHandler) ModuleCallsHandler(ctx context.Context, args cmd.CommandArg
 		return response, err
 	}
 
-	found, _ := h.StateStore.Modules.ModuleByPath(modPath)
-	if found == nil {
-		return response, nil
-	}
-
-	if found.ModManifest == nil {
-		return response, nil
+	moduleCalls, err := h.StateStore.Modules.ModuleCalls(modPath)
+	if err != nil {
+		return response, err
 	}
 
-	response.ModuleCalls = h.parseModuleRecords(ctx, found.ModManifest.Records)
+	response.ModuleCalls = h.parseModuleRecords(ctx, moduleCalls)
 
 	return response, nil
 }
 
-func (h *CmdHandler) parseModuleRecords(ctx context.Context, records []datadir.ModuleRecord) []moduleCall {
-	// sort all records by key so that dependent modules are found
-	// after primary modules
-	sort.SliceStable(records, func(i, j int) bool {
-		return records[i].Key < records[j].Key
-	})
-
+func (h *CmdHandler) parseModuleRecords(ctx context.Context, moduleCalls tfmod.ModuleCalls) []moduleCall {
 	modules := make(map[string]moduleCall)
-	for _, manifest := range records {
-		if manifest.IsRoot() {
-			// this is the current directory, which is technically a module
-			// skipping as it's not relevant in the activity bar (yet?)
+	for _, manifest := range moduleCalls.Declared {
+		if manifest.SourceAddr == nil {
+			// We skip all modules without a source address
```

Comment:
Can you add a test for this case? On a related note I'm wondering if we should add the edge case of empty string to tfschema here https://github.com/hashicorp/terraform-schema/blob/43f5a813b1b10b271d39e7406c61b0eb807a5a9a/earlydecoder/load_module.go#L310-L318

Reply:
Since we now have a default case with `UnknownSourceAddr`, we might now be able to remove this check entirely. Before I was running into all kind of null-pointer exceptions. Maybe we can discuss this over :zoom: